tests/functional/conftest.py

```diff
@@ -0,0 +1,23 @@
+import subprocess
+
+import pytest
+import yaml
+
+
+@pytest.fixture(scope="session", autouse=True)
+def install_dcgm_snap():
+    """Install the snap and enable dcgm-exporter service for testing."""
+    with open("snap/snapcraft.yaml") as f:
+        snapcraft = yaml.safe_load(f)
+        snap_build_name = f"{snapcraft['name']}_*.snap"
+
+        subprocess.run(
+            f"sudo snap install --devmode {snap_build_name}",
```

Comment:
The snap is strict confinement, so we shouldn't use devmode - running in devmode can hide issues with confinement. 
 Instead you can use `--dangerous` to install the unsigned snap in strict confinement mode:

```suggestion
            f"sudo snap install --dangerous {snap_build_name}",
```